src/elm/Main.elm

```diff
@@ -859,6 +866,39 @@ goUp id (model, prevCmd) =
   model ! [prevCmd]
     |> activate targetId
 
+goToTopOfColumn : String -> (Model, Cmd Msg) -> (Model, Cmd Msg)
+goToTopOfColumn id (model, prevCmd) =
+  let
+      firstSibling = getSiblings id model.workingTree.tree
+        |> ListExtra.getAt 0
+
+      targetId = case firstSibling of
+        Nothing -> id
+        Just firstSiblingTree -> firstSiblingTree.id
+
+  in
+  model ! [prevCmd]
+    |> activate targetId
+
+goToBottomOfColumn : String -> (Model, Cmd Msg) -> (Model, Cmd Msg)
+goToBottomOfColumn id (model, prevCmd) =
+  let
+    siblings = getSiblings id model.workingTree.tree
+    siblingCount =
+      siblings
+        |> List.length
+
+    lastSiblingTree = getSiblings id model.workingTree.tree
+      |> ListExtra.getAt (siblingCount-1)
```

Comment:
Elm uses `List` a lot, as you may have noticed. It's a good time investment to learn the [core List functions](http://package.elm-lang.org/packages/elm-lang/core/5.1.1/List) well.

For instance, this line can (and I think, should) be replaced with
```elm
   |> List.reverse
   |> List.head
```
Then you don't need to include the added `ListExtra` library, nor do you need to get the siblingCount. Same goes for `ListExtra.getAt 0` above.

Reply:
For other elements, then in that case no I'd probably use List.Extra package